dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -15,121 +15,330 @@
+      if (is_constant != other.is_constant) {
+        return !is_constant;  // Non-constants sort before constants
+      }
+      return is_constant ? frame_num < other.frame_num : frame.frame_idx < other.frame.frame_idx;
+    }
+
+    int frame_num;
+    bool is_constant = false;
+    union {
+      struct {
+        int frame_idx;
+      } frame;
+      struct {
+        uint8_t value;
+      } constant;
+    };
+  };
+
  public:
+  USE_OPERATOR_MEMBERS();
   using InBackend = CPUBackend;
-  using OutBackend = std::conditional_t<std::is_same_v<Backend, CPUBackend>,
-                                        CPUBackend,
-                                        GPUBackend>;
+  using OutBackend =
+      std::conditional_t<std::is_same_v<Backend, CPUBackend>, CPUBackend, GPUBackend>;
+  using OutStorage = detail::storage_tag_map_t<OutBackend>;
+
+  explicit VideoDecoderBase(const OpSpec &spec) : Operator<Backend>(spec) {
+    if (spec_.HasArgument("device")) {
+      auto device_str = spec_.template GetArgument<std::string>("device");
+      if (device_str == "mixed") {
+        thread_pool_ = std::make_unique<ThreadPool>(
+            spec.GetArgument<int>("num_threads"), spec.GetArgument<int>("device_id"),
+            spec.GetArgument<bool>("affine"), "VideoDecoder");
+      }
+    }
+
+    DALI_ENFORCE(!(frames_.HasValue() &&
+                   (start_frame_.HasValue() || stride_.HasValue() || sequence_length_.HasValue())),
+                 "Cannot specify both `frames` and any of `start_frame`, `sequence_length`, "
+                 "`stride` arguments");
 
-  VideoDecoderBase() = default;
+    if (frames_.HasValue() && (spec_.HasArgument("pad_mode") || spec_.HasArgument("pad_value"))) {
+      DALI_WARN(
+          "Padding options (`pad_mode`, `pad_value`) are ignored when `frames` argument is "
+          "provided");
+    }
+
+    auto pad_mode_str = spec_.template GetArgument<std::string>("pad_mode");
+    if (pad_mode_str == "none" || pad_mode_str == "") {
+      boundary_type_ = boundary::BoundaryType::ISOLATED;
+    } else if (pad_mode_str == "constant") {
+      boundary_type_ = boundary::BoundaryType::CONSTANT;
+    } else if (pad_mode_str == "edge") {
+      boundary_type_ = boundary::BoundaryType::CLAMP;
+    } else if (pad_mode_str == "reflect_1001" || pad_mode_str == "symmetric") {
+      boundary_type_ = boundary::BoundaryType::REFLECT_1001;
+    } else if (pad_mode_str == "reflect_101" || pad_mode_str == "reflect") {
+      boundary_type_ = boundary::BoundaryType::REFLECT_101;
+    } else {
+      DALI_FAIL(make_string("Invalid pad_mode: ", pad_mode_str, "\n",
+                            "Valid options are: none, constant, edge, reflect_1001, reflect_101"));
+    }
+
+    if (boundary_type_ == boundary::BoundaryType::CONSTANT) {
+      pad_value_ = spec_.template GetArgument<int>("pad_value");
+      DALI_ENFORCE(pad_value_ >= 0 && pad_value_ <= 255, "pad_value must be in range [0, 255]");
```

Comment:
Is there any value in extending that to provide <C> or <R, G, B> to have any fill value available?

Reply:
Will do